solutions/0200-0299/0289-game-of-life-medium.md

```diff
@@ -0,0 +1,132 @@
+---
+description: 'Author: @Pranathi-star | https://leetcode.com/problems/game-of-life/'
+tags: [2-D Array, Iteration]
+---
+
+# 0289 - Game of Life
+
+## Problem Statement
+
+According to Wikipedia's article: "**The Game of Life**, also known simply as **Life**, is a cellular automaton devised by the British mathematician John Horton Conway in 1970."
+
+The board is made up of an `m x n` grid of cells, where each cell has an initial state: **live** (represented by a `1`) or **dead** (represented by a `0`). Each cell interacts with its eight neighbors (horizontal, vertical, diagonal) using the following four rules (taken from the above Wikipedia article):
+
+- Any live cell with fewer than two live neighbors dies as if caused by under-population.
+- Any live cell with two or three live neighbors lives on to the next generation.
+- Any live cell with more than three live neighbors dies, as if by over-population.
+- Any dead cell with exactly three live neighbors becomes a live cell, as if by reproduction.
+The next state is created by applying the above rules simultaneously to every cell in the current state, where births and deaths occur simultaneously. Given the current state of the `m x n` grid `board`, return the next state.
+
+**Example 1:**
+
+```
+Input: board = [[0,1,0],[0,0,1],[1,1,1],[0,0,0]]
+Output: [[0,0,0],[1,0,1],[0,1,1],[0,1,0]]
+```
+
+**Example 2:**
+
+```
+Input: board = [[1,1],[1,0]]
+Output: [[1,1],[1,1]]
+```
+
+
+**Constraints:**
+
+- `m == board.length`
+- `n == board[i].length`
+- `1 <= m, n <= 25`
+- `board[i][j] is 0 or 1.`
+
+## Approach 1: Iterating the 2-D array
+
+<Tabs>
+<TabItem value="py" label="Python">
+<SolutionAuthor name="@pranathi_kodich"/>
+
+```py
+# Time Complexity: O(R * C). 
+# Each cell in the matrix is visited atmost once
+# Space Complexity: O(1). 
+# Constant space required for storing directions.
+
+class Solution:
```

Comment:
Format your code well please. The indentation looks awful in code. Even tho it looks ok in UI.